### PR TITLE
SB extension API updates

### LINKED ISSUE #21138: [BUG] Binding to JObject doesn't work correctly in Service Bus extension
When creating a function that binds to a JObject, if the received message does not have a contentType of application/json, a stackoverflow occurs - apparently due to the DataContractBinarySerializer getting confused. A simple fix would be to reorder the operations listed [here](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/MessageToPocoConverter.cs#L24-L26), so that we do the Json conversion attempt before the XML attempt. Or we could check the typeof(TElement) and special case JObject to preserve the existing behavior as much as possible.

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/21138

### CODE REVIEW

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Bindings/AsyncCollectorArgumentBindingProvider.cs [hunk 1]
```diff
@@ -33,22 +40,22 @@ public IArgumentBinding<ServiceBusEntity> TryCreate(ParameterInfo parameter)
             return CreateBinding(itemType);
         }
 
-        private static IArgumentBinding<ServiceBusEntity> CreateBinding(Type itemType)
+        private IArgumentBinding<ServiceBusEntity> CreateBinding(Type itemType)
         {
             MethodInfo method = typeof(AsyncCollectorArgumentBindingProvider).GetMethod("CreateBindingGeneric",
```
Comment (pakrym) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Bindings/AsyncCollectorArgumentBindingProvider.cs:45]: Ugh.

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Bindings/ServiceBusBinding.cs [hunk 2]
```diff
@@ -53,7 +53,8 @@ public async Task<IValueProvider> BindAsync(BindingContext context)
             var entity = new ServiceBusEntity
             {
                 MessageSender = messageSender,
-                EntityType = _attribute.EntityType
+                ServiceBusEntityType = _attribute.ServiceBusEntityType,
+                JsonSerializerSettings = _messagingProvider.Options.JsonSerializerSettings
```
Comment (pakrym) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Bindings/ServiceBusBinding.cs:57]: Why do we need `JsonSerializerSettings` on the `ServiceBusEntity`?

**Review by pakrym — APPROVED** [review]

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Bindings/ServiceBusBinding.cs (hunk 2 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Bindings/ServiceBusBinding.cs:57]: It's used in the UserTypeArgumentBindingProvider, but I will change to make it more consistent with other usages.

**Review by jsquire — APPROVED** [review]

**Pull request merged**